src/server.rs

```diff
@@ -29,3 +35,56 @@ async fn register_device(mut req: tide::Request<State>) -> tide::Result<tide::Re
 
     Ok(tide::Response::new(tide::StatusCode::Ok))
 }
+
+/// Notifies a single device with a visible notification.
+async fn notify_device(mut req: tide::Request<State>) -> tide::Result<tide::Response> {
+    let device_token = req.body_string().await?;
+    info!("Got direct notification for {device_token}.");
+
+    let (client, device_token) = if let Some(sandbox_token) = device_token.strip_prefix("sandbox:")
+    {
+        (req.state().sandbox_client(), sandbox_token)
+    } else {
+        (req.state().production_client(), device_token.as_str())
+    };
+
+    let db = req.state().db();
+    let payload = DefaultNotificationBuilder::new()
+        .set_title("New messages")
+        .set_title_loc_key("newmessage") // Localization key for the title.
+        .set_body("You could have new messages")
+        .set_loc_key("newmessagebody") // Localization key for the body.
```

Comment:
as discussed, we can strike the "could". 

for the localization keys - with chance, with these keys, translation already works for the title; will add one for the body

```suggestion
        .set_title_loc_key("new_messages") // Localization key for the title.
        .set_body("You have new messages")
        .set_loc_key("new_messages_body") // Localization key for the body.
```